src/common/SearchBar.js

```diff
@@ -0,0 +1,66 @@
+import Head from "next/head";
+import Image from "next/image";
+import { Inter } from "next/font/google";
+// import styles from "@/styles/Home.module.css";
+import FooterBar from "@/common/FooterBar";
+import ReusableNavLink from "@/common/ReusableNavLink";
+import Link from 'next/link';
+import Data from "../mock-data.json"
+import styles from "@/styles/SearchBar.module.css";
+import {useState} from 'react';
+
+
+function SearchBar({ placeholder, data }) {
+
+    const [filteredData, setFilteredData] = useState([]);
+
+    const handleFilter = (event) => {
+        const searchWord = event.target.value
+        const newFilter = data.filter((value) => {
+            return value.name.toLowerCase().includes(searchWord.toLowerCase());
+        });
+        if (searchWord === "") {
+            setFilteredData([])
+        } else {
+            setFilteredData(newFilter)
+        } 
+    };
+    return (
+        <div>
+            <form>
+                <div>
+                    <div className="search-inputs">
+                    <input id="search-input-bar" type="text" placeholder= {placeholder} name='search' required onChange={handleFilter} />
+                    <button type='submit'>
+                    <Image
+                        src="/search-glass.svg"
+                        alt="Search Image"
+                        className={styles.searchImage}
+                        width={20}
+                        height={20}
+                    />
+                    </button>
+                    </div>
+                </div>
+            </form>
+            {filteredData.length != 0 && (
```

Comment:
@annonymousauthority @jjleigh I am tagging you because there is some unwanted "expansion" in the NavBar when the SearchBar renders results. Please view attached GIF.

![jumpy-nav-bar-due-to-search-results](https://user-images.githubusercontent.com/32520970/235330121-c6dafe79-a320-4436-91b7-3f272003778b.gif)



Off the top of my head, I am thinking of having the search results appear "above" the rest of the elements on the page. However my attempts to add a `z-index` failed 🫠

Any ideas?

Reply:
I added comments to the codelines that might need updating